central/src/containers/RouteContainer.tsx

```diff
@@ -0,0 +1,298 @@
+    let updatedGame = {
+      id: game.id,
+      title: game.title,
+      description: game.description,
+      phaseOneTime: game.phaseOneTime,
+      phaseTwoTime: game.phaseTwoTime,
+      imageUrl: game.imageUrl,
+      grade: game.grade,
+      domain: game.domain,
+      cluster: game.cluster,
+      standard: game.standard,
+      questions: questions,
+    }
+    const result = await updateGame(updatedGame);
+    if (result) {
+      getSortedGames();
+    }
+    setAlert({ message: 'Game saved.', type: 'success' });
+  }
+
+  const handleQuestionBankClick = (gameDetails: any) => {
+    getQuestions();
+    history.push(`/gamemaker/${gameDetails.id}/addquestion`)
+  }
+
+  // @ts-ignore
+  const handleCloneGame = async (game) => {
+    const result = await cloneGame(game);
+    if (result) {
+      getSortedGames();
+      setAlert({ message: 'Game cloned.', type: 'success' });
+    }
+    return result
+  }
+
+  const handleDeleteGame = async (id: number) => {
+    const result = await deleteGames(id);
+    if (result) {
+      const games = sortGames(await fetchGames(), sortType);
+      // setGames(games);
+    }
+    setAlert({ message: 'Game deleted.', type: 'success' });
+  }
+
+  const handleDeleteQuestion = async (id: number, game: Game) => {
+    const result = await deleteQuestions(id)
+    if (result) {
+      getSortedGames()
+    }
+    setAlert({ message: 'Question deleted.', type: 'success' });
+  }
+
+  const handleUserAuth = (isAuth: boolean) => {
+    setIsUserAuth(isAuth);
+  }
+
+  const persistUserAuth = (async () => {
+    let user = null;
+    try {
+      user = await Auth.currentAuthenticatedUser();
+      let userSession = await Auth.userSession(user);
+      if (userSession) {
+        setIsUserAuth(true);
+      }
+    } catch (e) {
+      setIsUserAuth(false);
+    }
+  });
+
+  const getGames = async () => {
+    setLoading(true);
+    await getSortedGames();
+    setLoading(false);
+  };
+
+  const isResolutionMobile = useMediaQuery("(max-width: 760px)");
+
+  const handleSearchClick = (isClick: boolean) => {
+    setIsSearchClick(isClick);
+  }
+
+  const handleModalClose =(modalOpen: boolean) =>{ 
+    localStorage.setItem('userPlayedBefore', 'true');
+    setModalOpen(modalOpen);
+  };
+
+  const handleModalOpen = (modalOpen:boolean, showModalGetApp:boolean) => {
+    localStorage.setItem('userPlayedBefore', 'false');
+    setShowModalGetApp(showModalGetApp);
+    setModalOpen(modalOpen);
+  }
+
+  function checkUserPlayed(): boolean {
+    if (localStorage.getItem('userPlayedBefore') === 'true')
+      return false;
+    else 
+      return true;
+  };
+
+  const filterGame = (game: IGameTemplate | null, search: string) => {
+    if (!game || !game?.title) {
+      return false;
+    }
+    if (game.title.toLowerCase().indexOf(search) > -1 || (game.description && game.description.toLowerCase().indexOf(search) > -1)) {
+      return true;
+    }
```

Comment:
What's the purpose of this? If we're adding filtering for all games, we should make a server call to do so.

Reply:
I think this is used to handle the search bar. This is old code that's been moved to a new container. Agree that it needs to change.